Cargo.toml

```diff
@@ -92,6 +92,10 @@ name = "clock"
 [[bin]]
 name = "clone_macro"
 
+[[bin]]
+name = "communication_thread"
+edition = "2018"
```

Comment:
Unrelated to this PR but we should maybe move all the examples to 2018 at this point :) @GuillaumeGomez @EPashkin opinions?